tests/core/test_state_store.py

```diff
@@ -0,0 +1,89 @@
+# Standard Library
+import json
+import os
+import shutil
+import time
+
+# First Party
+from smdebug.core.state_store import StateStore
+
+
+def setup_test():
+    try:
+        shutil.rmtree("checkpoints_test_dir/")
+    except:
+        pass
+    # create the checkpoints directory
+    os.mkdir("checkpoints_test_dir/")
+    dir_path = os.path.abspath("checkpoints_test_dir")
+
+    # create the config file and set the corresponding environment variable.
+    mock_config = {"LocalPath": dir_path}
+    with open("mock_config.json", "w") as f:
+        json.dump(mock_config, f)
+    os.environ["CHECKPOINT_CONFIG_FILE_PATH"] = os.path.abspath("mock_config.json")
+
+    # create the metadata file inside the checkpoints directory.
+    mock_metadata = [
+        {
+            "training-run": "",
+            "latest-global-step-saved": "",
+            "latest-global-step-seen": "",
+            "latest-mode-step": "",
+        }
+    ]
+    with open(dir_path + "/metadata.json", "w") as f:
+        json.dump(mock_metadata, f)
+    return dir_path, os.path.abspath("mock_config.json")
+
+
+def cleanup(checkpoints_dir_path, config_path):
+    shutil.rmtree(checkpoints_dir_path)
+    os.remove(config_path)
+
+
+def test_is_checkpoint_updated():
+    os.environ["SMDEBUG_LOG_LEVEL"] = "debug"
+    s1 = StateStore()
+    # There is no checkpoint_dir. is_checkpoint_updated should return False.
+    assert s1.is_checkpoint_updated() is False
+
+    # call setup_test to create checkpoints_dir and metadata file.
+    checkpoints_dir_path, config_path = setup_test()
+    s2 = StateStore()
+    # checkpoints_dir only has metadata.json. So no checkpoints file was created or updated. It should return false.
+    assert s2.is_checkpoint_updated() is False
+
+    s2.update_state("test-state1")
+    time.sleep(1)
+    # checkpoints_dir still has only metadata.json. It should return false.
+    assert s2.is_checkpoint_updated() is False
+
+    os.mkdir(s2._checkpoint_dir + "/subdir1")
+    with open(s2._checkpoint_dir + "/subdir1/checkpoint_test1.txt", "w") as f:
+        f.write("checkpoint-test-string-1")
```

Comment:
sync here . f.flush() to make sure that next line is certain to return true.

Reply:
flush here might not make any difference. 

We could probably remove the sleep if we did a `fileObject.flush` and `os.fsync(f)` here - [state_store.py#L104](https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/core/state_store.py#L104)

Reply:
yeah, this is what I suggested earlier, please remove the sleep from code and use flush(method which does fsync) . we dont neeed to do fsync in code, but in test